### ISSUE TITLE
avoid wb.fullname complexity if not needed on "Book(filename)"

### ISSUE THREAD

**Issue by sdementen** [xlwings/xlwings#1842]
If one provides just the name of the workbook (not the fullname) to get a reference to an opened workbook, it should be possible to avoid relying on the "not yet 100% robust" sharepoint/local name resolution.
It is a matter of swapping the order of the tests in https://github.com/xlwings/xlwings/blob/main/xlwings/main.py#L823 
from 
```
                        if (
                            wb.fullname.lower() == fullname
                            or wb.name.lower() == fullname
                        ):
```
to
```
                        if (
                            wb.name.lower() == fullname
                            or wb.fullname.lower() == fullname
                        ):
```

**Comment by fzumstein** [comment]
yes, that should work, similar to this one: https://github.com/xlwings/xlwings/commit/55034131235b4e674ebc5ea938f5c18839d53775

**Comment by sdementen** [comment]
In fact, if multiple files are open, there will still be calls to .fullname which raises exception. A try/except must be added around the .fullname :-(

**Comment by sdementen** [comment]
The .fullname logic fails when a subfolder of a SharePoint is synchronized locally

**Comment by fzumstein** [comment]
this should all go away though with https://github.com/xlwings/xlwings/issues/1829 ?

**Comment by fzumstein** [comment]
Trying this out, it's actually unclear to me what use case this fixes?

**Comment by fzumstein** [comment]
see also https://github.com/xlwings/xlwings/issues/1873

**Comment by fzumstein** [comment]
I am closing this as I believe this should be handled by 

* https://github.com/xlwings/xlwings/issues/1873 and
* https://github.com/xlwings/xlwings/issues/1829

If not, feel free to reopen (the fixes will be released with 0.27.4).
